examples/Microsoft.Spark.CSharp.Examples/Sql/Streaming/README.md

```diff
@@ -0,0 +1,112 @@
+# .NET for Apache Spark C# Samples: Streaming
+
+[.NET for Apache Spark](https://dot.net/spark) is a free, open-source, and cross-platform big data analytics framework.
+
+In the **Streaming** folder, we provide C# samples which will help you get started with the big data analytics scenario known as
+**structured streaming.** Stream processing means we're analyzing data live as it's being produced.
+
+## Problem
+
+These samples are examples of **stream processing** since we're processing data in real time.
+
+## Solution
+
+### 1. Create a Spark Session
+
+In any Spark application, we need to establish a new `SparkSession`, which is the entry point to programming Spark with the Dataset and DataFrame API.
+
+```CSharp
+SparkSession spark = SparkSession
+       .Builder()
+       .AppName("My Streaming App")
+       .GetOrCreate();
+```
+
+By calling on the *spark* object created above, we can access Spark and DataFrame functionality throughout our program.
+
+### 2. Establish and Connect to Data Stream
+
+#### Establish Stream: Netcat
+
+netcat (also known as *nc*) allows you to read from and write to network connections. We'll establish a network
+connection with netcat through a terminal window.
+
+[Download netcat](https://sourceforge.net/projects/nc110/files/), extract the file from the zip download, and append the 
+directory you extracted to your "PATH" environment variable.
+
+To start a new connection, open a command prompt. For Linux users, run ```nc -lk 9999``` to connect to localhost on port 9999.
+
+Windows users can run ```nc -vvv -l -p 9999``` to connect to localhost port 9999. The result should look something like this:
+
+![NetcatConnect](https://github.com/bamurtaugh/spark/blob/StreamingLog/examples/Microsoft.Spark.CSharp.Examples/Sql/Streaming/netconnect.PNG)
+
+Our Spark program will be listening for input we type into this command prompt.
+
+#### Connect to Stream: ReadStream()
+
+The `ReadStream()` method returns a `DataStreamReader` that can be used to read streaming data in as a `DataFrame`. We'll include the host and port information so that our Spark app knows where to expect its streaming data.
+
+```CSharp
+DataFrame words = spark
+      .ReadStream()
+      .Format("socket")
+      .Option("host", hostname)
+      .Option("port", port)
+      .Load();
+```
+
+### 3. Register a UDF
+
+A UDF is a *user-defined function.* We can use UDFs in Spark applications to perform calculations and analysis on our data.
+
+```CSharp
+spark.Udf().Register<string, bool>("MyUDF", input => ValidLogTest(input));
```

Comment:
Is `ValidLogTest` definition anywhere?  Is it a black box?  Do you want to describe what it is supposed to do ?

Reply:
I wanted to show an example of using a UDF on streaming data. I had originally created a streaming log processing example (hence `ValidLogTest`), but then I moved away from that example and created a different sample for my videos/conference demo. I have now uploaded that sample instead: `StructuredNetworkWordCountUDF.cs`, and I'll reference that one for things like the UDF in the readme.